### PR TITLE
Parsing physical rack

### PR DESCRIPTION
**Pull request by felipedf**
Parse info from a PhysicalRack

 * To be merged with: ManageIQ/manageiq#16853

### CODE REVIEW

**Comment by felipedf** [comment]
The Travis error is due the server[:rack] field that is not delete after parsing, this is done here ManageIQ/manageiq#16853 and will be ok after merge. @agrare

**Comment by rodneyhbrown7** [comment]
Can you rebase this with master?

**Comment by felipedf** [comment]
Done

**Comment by felipedf** [comment]
@miq-bot assign @agrare

**Comment by agrare** [comment]
@felipedf can you add some tests to the [full refresh test ](https://github.com/felipedf/manageiq-providers-lenovo/blob/082a09a73665d0add56001357e93672cdba64bbd/spec/models/manageiq/providers/lenovo/physical_infra_manager/refresher_spec.rb#L68) to check for the physical racks and the association between the physical servers and the physical racks?

Convention here would be to add a `assert_specific_rack` to check for a specific rack's attributes and a `assert_specific_server` where you can check the server has a relationship back to the rack.

File: app/models/manageiq/providers/lenovo/physical_infra_manager.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 class ManageIQ::Providers::Lenovo::PhysicalInfraManager < ManageIQ::Providers::PhysicalInfraManager
-  has_many :physical_servers, :foreign_key => "ems_id", :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalServer", :dependent => :destroy
-
+  has_many :physical_racks, :foreign_key => "ems_id", :dependent => :destroy, :inverse_of => :ext_management_system,
+           :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalRack"
+  has_many :physical_servers, :foreign_key => "ems_id", :dependent => :destroy, :inverse_of => :ext_management_system,
+           :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalServer"
```
Comment (agrare) [inline app/models/manageiq/providers/lenovo/physical_infra_manager.rb]: Shouldn't the physical_racks and physical_servers associations be on the core physical_infra_manager?  And you shouldn't need the class name as long as the assoc name matches the base class name

File: app/models/manageiq/providers/lenovo/physical_infra_manager/physical_server.rb [hunk 2]
```diff
@@ -2,6 +2,12 @@ module ManageIQ::Providers
   class Lenovo::PhysicalInfraManager::PhysicalServer < ::PhysicalServer
     include_concern 'RemoteConsole'
 
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :inverse_of => :physical_servers,
+               :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager"
+
+    belongs_to :physical_rack, :dependent => :destroy, :inverse_of => :physical_servers,
+               :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalRack"
```
Comment (agrare) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/physical_server.rb]: Again I think all of these associations should be defined on the base models without the class_name, not here

File: app/models/manageiq/providers/lenovo/physical_infra_manager/physical_rack.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module ManageIQ::Providers
+  class Lenovo::PhysicalInfraManager::PhysicalRack < ::PhysicalRack
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :inverse_of => :physical_racks,
+               :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager"
+
+    has_many :physical_servers, :dependent => :destroy, :inverse_of => :physical_rack,
+             :class_name => "ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalServer"
```
Comment (agrare) [inline app/models/manageiq/providers/lenovo/physical_infra_manager/physical_rack.rb]: Same this should be in the base model

**Pull request merged**